cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MetadataDataset.java

```diff
@@ -1091,6 +1091,7 @@ private Metadata writeWithHistory(Metadata existing, MetadataEntry entry, Set<In
       }
       if (isNewEntity) {
         indexers.add(new MetadataEntityTypeIndexer());
+        indexers.add(new NearestKnownAncestorIndexer());
```

Comment:
this will also index known entities as themselves right?

Reply:
if a known entity is passed to NearestKnownAncestorIndexer then it returns an empty set.